clouddrift/signal.py

```diff
@@ -234,6 +234,190 @@ def cartesian_to_rotary(
     return wp, wn
 
 
+def ellipse_parameters(
+    xa: Union[np.ndarray, xr.DataArray],
+    ya: Union[np.ndarray, xr.DataArray],
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """Return the instantaneous parameters of a modulated elliptical signal from its analytic Cartesian signals.
+
+    Parameters
+    ----------
+    xa : array_like
+        Complex-valued analytic signal for first Cartesian component (zonal, east-west).
+    ya : array_like
+        Complex-valued analytic signal for second Cartesian component (meridional, north-south).
+
+    Returns
+    -------
+    kappa : np.ndarray
+        Ellipse root-mean-square amplitude.
+    lambda : np.ndarray
+        Ellipse linearity between -1 and 1, or departure from circular motion (lambda=0).
+    theta : np.ndarray
+        Ellipse orientation in radian.
+    phi : np.ndarray
+        Ellipse phase in radian.
+
+    Examples
+    --------
+
+    To obtain the ellipse parameters from a pair of real-valued signals (x, y):
+
+    >>> kappa, lambda, theta, phi = ellipse_parameters(analytic_signal(x), analytic_signal(y))
```

Comment:
Maybe add one simple example with a few short input arrays (1, 2, 3.. or similar?) and show what the output looks like.

Reply:
I wish I could but it does not really work for short inputs